packages/amplication-git-service/src/providers/githubProvider.ts

```diff
@@ -0,0 +1,323 @@
+          repo: this.repo,
+          title,
+          body,
+          base,
+          head: branch
+        })
+      ).data;
+      return {
+        url,
+        number
+      };
+    } catch (err) {
+      switch (err?.response?.data?.errors[0]?.message) {
+        case `No commits between ${base.toLowerCase()} and ${branch.toLowerCase()}`:
+          throw new Error(`There is no new commit in branch ${branch}`);
+        case `A pull request already exists for ${this.owner}:${branch}.`:
+          throw new Error('Branch already has pull request');
+        default:
+          throw err;
+      }
+    }
+  }
+
+  public async addPullRequestComment(
+    pullRequestNumber: number,
+    body: string
+  ): Promise<string> {
+    try {
+      const { html_url } = (
+        await this.octokit.rest.issues.createComment({
+          owner: this.owner,
+          repo: this.repo,
+          issue_number: pullRequestNumber,
+          body
+        })
+      ).data;
+      return html_url;
+    } catch (err) {
+      switch (err.message) {
+        case 'Not Found':
+          throw new Error(
+            'Cannot add comment to pull request: pull request not found'
+          );
+        default:
+          console.log(err);
+          throw err;
+      }
+    }
+  }
+
+  public async getOpenedPullRequest(branch: string): Promise<PullRequestMeta> {
+    try {
+      const [pullRequest] = (
+        await this.octokit.rest.pulls.list({
+          owner: this.owner,
+          repo: this.repo,
+          head: `${this.owner}:${branch}`,
+          state: 'open'
+        })
+      ).data;
+      if (pullRequest) {
+        return {
+          url: pullRequest.html_url,
+          number: pullRequest.number
+        };
+      } else {
+        return null;
+      }
+    } catch (err) {
+      switch (err.message) {
+        default:
+          throw err;
+      }
+    }
+  }
+
+  public async updatePullRequest(
+    number: number,
+    state: boolean
+  ): Promise<void> {
+    await this.octokit.rest.pulls.update({
+      owner: this.owner,
+      repo: this.repo,
+      pull_number: number,
+      state: state ? 'open' : 'closed'
+    });
+  }
+
+  public async createBranch(
+    branch: string,
+    headCommit: string
+  ): Promise<string> {
+    try {
+      return (
+        await this.octokit.rest.git.createRef({
+          owner: this.owner,
+          repo: this.repo,
+          ref: `refs/heads/${branch}`,
+          sha: headCommit
+        })
+      ).data.ref;
+    } catch (err) {
+      switch (err.message) {
+        case 'Reference already exists':
+          return null;
+        default:
+          throw err;
+      }
+    }
+  }
+
+  public async deleteBranch(branch: string): Promise<boolean> {
+    try {
+      const { status } = await this.octokit.rest.git.deleteRef({
+        owner: this.owner,
+        repo: this.repo,
+        ref: `heads/${branch}`
+      });
+      return status === 204;
+    } catch (exception) {
```

Comment:
Let's be consistent with `catch()` argument names. Here we call it `exception`, above we called it `err`.

Can you please tackle it in all other places as well?